doc/dev/design/FABRID.rst

```diff
@@ -0,0 +1,382 @@
+        &HVF_i = MAC_{K_i}(Identifier, ingress_i, egress_i, encryptedPolicyID_i,\\& srcAddrLen, srcHostAddr)[0:3] \oplus 0x3FFFFF\\\\
+        &HVFVerified_i = MAC_{K_i}(Identifier, ingress_i, egress_i, encryptedPolicyID_i,\\& srcAddrLen, srcHostAddr)[3:6] \oplus 0x3FFFFF\\\\
+    \end{align*}
+
+Data plane
+----------
+
+Processing at the router
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Whenever a FABRID enabled router receives a SCION packet, it has to figure out whether it should be processed as FABRID or not.
+In both cases, all the logic of a normal SCION packet will be applied too.
+The router determines whether the SCION packet is a FABRID packet as follows:
+
+.. image:: fig/FABRID/FABRIDActivation.png
+    :scale: 70%
+
+If the SCION packet uses FABRID, the router is going to verify the correctness of the current FABRID Hop-validation-field using
+either the AS-to-AS or AS-to-Host DRKey and verifies whether the encrypted policy index matches a valid FABRID policy.
+If this is the case, the router will update the FABRID Hop-validation-field accordingly and route the packet over
+an intra-AS path matching the provided FABRID policy.
+The corresponding intra-AS paths are provided to the border routers by the local control service.
+
+Processing at the endhost
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To be able to send a FABRID packet, the endhost has to choose a path that supports its path constraints.
+Then it can request the necessary DRKeys from its local control service.
+With this the endhost is able to create FABRID packets and then send them to the border router for further forwarding.
+The receiving endhost can then recompute the path validator to verify that the packet was forwarded over this path.
+The FABRID snet implementation will automatically request the necessary DRKeys and compute the hop validation fields.
+The endhost only has to provide the path and the FABRID policies.
+
+Control plane
+---------------
+
+Control service
+^^^^^^^^^^^^^^^^^
+
+The control plane for FABRID is responsible for parsing FABRID policies into corresponding data structures.
+Through gRPC, border routers can query the control service for the list of supported policies per interface,
+as well as the mapping from policies to MPLS labels.
+Policies are disseminated to remote ASes through PCBs, which clients in the AS can query from their Path Servers.
+This policy information can also be requested directly from remote ASes over gRPC.
+
+The control service introduces a FABRID service with the following endpoints where intra-AS means it can be reached
+from the local AS and inter-AS means it can be reached from a remote AS:
+
+- GetMPLSMapIfNecessary (intra-AS)
+- GetRemotePolicyDescription (intra-AS)
+- GetSupportedIndicesMap (inter-AS, intra-AS)
+- GetIndexIdentifierMap (inter-AS, intra-AS)
+- GetLocalPolicyDescription (inter-AS, intra-AS)
+
+Important data structures
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The FABRID service uses the following important data structures:
```

Comment:
The following list explains the most important data structures used in the FABRID service: